src/garage/torch/modules/mlp_module.py

```diff
@@ -0,0 +1,91 @@
+            None to maintain a linear activation.
+        hidden_w_init (callable): Initializer function for the weight
+            of intermediate dense layer(s). The function should return a
+            tf.Tensor.
+        hidden_b_init (callable): Initializer function for the bias
+            of intermediate dense layer(s). The function should return a
+            tf.Tensor.
+        output_nonlinearity (callable): Activation function for output dense
+            layer. It should return a tf.Tensor. Set it to None to
+            maintain a linear activation.
+        output_w_init (callable): Initializer function for the weight
+            of output dense layer(s). The function should return a
+            tf.Tensor.
+        output_b_init (callable): Initializer function for the bias
+            of output dense layer(s). The function should return a
+            tf.Tensor.
+        layer_normalization (bool): Bool for using layer normalization or not.
+    """
+    def __init__(self,
+                 input_dim,
+                 output_dim,
+                 hidden_sizes,
+                 hidden_nonlinearity=nn.ReLU,
+                 hidden_w_init=nn.init.xavier_uniform_,
+                 hidden_b_init=nn.init.zeros_,
+                 output_nonlinearity=None,
+                 output_w_init=nn.init.xavier_uniform_,
+                 output_b_init=nn.init.zeros_,
+                 layer_normalization=False):
+        super(MlpModule, self).__init__()
+
+        self._input_dim = input_dim
+        self._output_dim = output_dim
+        self._hidden_sizes = hidden_sizes
+        self._hidden_nonlinearity  = hidden_nonlinearity
+        self._hidden_w_init = hidden_w_init
+        self._hidden_b_init = hidden_b_init
+        self._output_nonlinearity = output_nonlinearity
+        self._output_w_init = output_w_init
+        self._output_b_init = output_b_init
+        self._layer_normalization = layer_normalization
+
+        self.layers = []
+
+        # Hidden self.layers
+        in_dim = input_dim
+        for i, out_dim in enumerate(hidden_sizes):
+            hidden_layer = nn.Linear(in_dim, out_dim)
+            hidden_w_init(hidden_layer.weight)
+            hidden_b_init(hidden_layer.bias)
+
+            if layer_normalization:
+                ln = nn.LayerNorm(in_dim)
+                self.layers.append(ln)
+
+            self.layers.append(hidden_layer)
+            if hidden_nonlinearity:
+                self.layers.append(hidden_nonlinearity())
+
+            in_dim = out_dim
+
+        # Output self.layers
+        output_layer = nn.Linear(in_dim, output_dim)
+        output_w_init(output_layer.weight)
+        output_b_init(output_layer.bias)
+        self.layers.append(output_layer)
+
+        if output_nonlinearity is not None:
+            self.layers.append(output_nonlinearity())
+
+    def forward(self, inputs):
+        return nn.Sequential(*self.layers)(inputs)
```

Comment:
Please do not create a new module to do this. In this case a new module will be created for every forward pass.

Reply:
I'll remove the MlpModule and use Linda's MlpModule